dataset/dataset_operations_common.h

```diff
@@ -45,51 +45,23 @@ template <class T> auto copy_map(const T &map) {
   return out;
 }
 
-static inline void expectAlignedCoord(const Dim coord_dim, const Variable &var,
-                                      const Dim operation_dim) {
-  // Coordinate is 2D, but the dimension associated with the coordinate is
-  // different from that of the operation. Note we do not account for the
-  // possibility that the coordinates actually align along the operation
-  // dimension.
-  if (var.dims().ndim() > 1 && var.dims().contains(operation_dim))
-    throw except::DimensionError("Coordinate " + to_string(coord_dim) +
-                                 " contains the operation dim " +
-                                 to_string(operation_dim) +
-                                 ", but has more than one dimension. It will "
-                                 "thus not be reduced by the operation.");
-}
-
 template <bool ApplyToData, class Func, class... Args>
 DataArray apply_or_copy_dim_impl(const DataArray &a, Func func, const Dim dim,
                                  Args &&... args) {
-  const auto coord_apply_or_copy_dim = [&](auto &coords_, const auto &view,
-                                           const bool aligned) {
-    // Note the `copy` call, ensuring that the return value of the ternary
-    // operator can be moved. Without `copy`, the result of `func` is always
-    // copied.
+  const auto copy_independent = [&](auto &coords_, const auto &view,
+                                    const bool share) {
     for (auto &&[d, coord] : view)
-      if (coord.dims().ndim() == 0 || dim_of_coord(coord, d) != dim) {
-        if (aligned)
-          expectAlignedCoord(d, coord, dim);
-        if constexpr (ApplyToData) {
-          coords_.emplace(d, coord.dims().contains(dim)
-                                 ? func(coord, dim, args...)
```

Comment:
I think this line was actually unreachable, indicating that old implementation was too complex.